core/trino-main/src/main/java/io/trino/cost/ValuesStatsRule.java

```diff
@@ -38,9 +38,9 @@
 import static com.google.common.collect.ImmutableList.toImmutableList;
```

Comment:
I am not sure if it is a good enough rationale to make things public in SPI. Once things get there they won't be able to escape. Can we fix the loop without changing SPI? 

Notice unknown type is used mostly in analysis, a connector should always return known types.

Reply:
fixed by https://github.com/trinodb/trino/pull/15312#discussion_r1040836832